test/unit/options_validator_tests.js

```diff
@@ -445,4 +447,123 @@ describe('Options Validation', function() {
     const validatedObject = testClass.testOperation(testObject);
     expect(validatedObject).to.deep.equal({ a: 'override' });
   });
+
+  it('Should properly validate when no options are provided', function() {
+    class TestClass {
+      constructor() {
+        this.s = { options: { validationLevel: 'error' } };
+      }
+    }
+
+    TestClass.prototype.testOperation = arityOne()
+      .options({ a: { type: 'string' } })
+      .build(function(requiredArgument) {
+        return requiredArgument;
+      });
+
+    TestClass.prototype.testOperationTwo = arityTwo()
+      .options({ a: { type: 'string' } })
+      .build(function(requiredArgument, requiredArgumentTwo) {
+        return requiredArgument + requiredArgumentTwo;
+      });
+
+    const testClass = new TestClass();
+    const testResult = testClass.testOperation(1);
+    expect(testResult).to.equal(1);
+
+    const testResultTwo = testClass.testOperationTwo(1, 2);
+    expect(testResultTwo).to.equal(3);
+  });
+
+  it('Should fail with an object in the options position', function() {
+    class TestClass {
+      constructor() {
+        this.s = { options: { validationLevel: 'error' } };
+      }
+    }
+
+    TestClass.prototype.testOperation = arityOne()
+      .options({})
+      .build(function(requiredArgument) {
+        return requiredArgument;
+      });
+
+    const errorMessage = 'This operation has a required arity of 1, but 4 arguments were provided.';
+    const testClass = new TestClass();
+    expect(() => {
+      testClass.testOperation({ a: 1 }, { b: 2 }, () => {}, () => {});
+    }).to.throw(errorMessage);
+  });
+
+  it('Should correctly handle a promise', function() {
+    class TestClass {
+      constructor() {
+        this.s = { options: { validationLevel: 'error' } };
+      }
+    }
+
+    TestClass.prototype.testOperation = arityOne()
+      .options({ b: { type: 'boolean', default: false } })
+      .build(function(requiredArgument, options, callback) {
+        return this.testOperation2(requiredArgument, options, callback);
+      });
+
+    TestClass.prototype.testOperation2 = arityOne()
+      .options({ b: { type: 'boolean' } })
+      .build(function(requiredArgument, options, callback) {
+        return executeTestOperation(testOperation2, [this, requiredArgument, options, callback]);
```

Comment:
can't you just use `executeOperation` from `utils.js` here, instead of implementing part of it in the test file?

Reply:
I was trying to avoid the `topology` and `session` logic inside of `executeOperation`.

Reply:
In that case, I think you're just trying to test a method that returns a `Promise` after executing right? Can't you just do:
```
function executeTestOperation(operation, args) {
  return new Promise(function(resolve, reject) {
    args[args.length - 1] = (err, result) => err ? reject(err) : resolve(result);;
    return operation.apply(null, args);
  });
}
```